source/solvers/vof.cc

```diff
@@ -1082,5 +1116,219 @@ VolumeOfFluid<dim>::assemble_mass_matrix_diagonal(
     }
 }
 
+template <int dim>
+template <typename VectorType>
+void
+VolumeOfFluid<dim>::apply_peeling_wetting(
+  const unsigned int i_bc,
+  const VectorType & current_solution_cfd)
+{
+  const DoFHandler<dim> *dof_handler_cfd =
+    multiphysics->get_dof_handler(PhysicsID::fluid_dynamics);
+
+  // Initializations
+  locally_owned_dofs = dof_handler.locally_owned_dofs();
+  std::vector<types::global_dof_index> dof_indices_vof(
+    fe->dofs_per_cell); //  local connectivity
+
+  solution_pw.reinit(locally_owned_dofs, triangulation->get_communicator());
+  solution_pw = present_solution;
+
+  FEFaceValues<dim> fe_face_values_vof(*this->fs_mapping,
+                                       *this->fe,
+                                       *this->face_quadrature,
+                                       update_values |
+                                         update_quadrature_points);
+
+  FEFaceValues<dim> fe_face_values_cfd(*this->fs_mapping,
+                                       dof_handler_cfd->get_fe(),
+                                       *this->face_quadrature,
+                                       update_values |
+                                         update_quadrature_points);
+
+  const unsigned int n_q_points = this->cell_quadrature->size();
+
+  const FEValuesExtractors::Scalar pressure(dim);
+  std::vector<double>              pressure_values(n_q_points);
+  std::vector<double>              phase_values(n_q_points);
+
+  unsigned int boundary_id =
+    this->simulation_parameters.boundary_conditions.id[i_bc];
+
+  // Useful definitions for readability
+  const unsigned int density_fluid_0 =
```

Comment:
why are the types of these values unsigned int?

Reply:
it should absolutely not! Thank you for spotting this sloppy copy/paste ^^"